core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3117,6 +3117,14 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     val responseData = new LiCombinedControlResponseData()
 
+    decomposedRequest.updateMetadataRequest match {
```

Comment:
I think we should have a comment here to describe the case that LeaderAndIsr may depend on updateMetadataRequest.

Reply:
Thanks for the review. I've added a comment.